crates/pipeline-manager/src/compiler/rust_compiler.rs

```diff
@@ -1094,209 +1158,207 @@ async fn cleanup_rust_compilation(
             }
         };
 
-    // (1) Retrieve existing pipeline programs
-    //     (pipeline_id, program_version, program_binary_source_checksum, program_binary_integrity_checksum)
-    let existing_pipeline_programs = db
-        .lock()
-        .await
-        .list_pipeline_programs_across_all_tenants()
-        .await?;
-
-    // For compilation-related cleanup, only the source checksum is needed
-    let mut existing_source_checksums: Vec<String> = existing_pipeline_programs
+    // Determine all the crates and their artifact names that are currently in use
+    let mut crates_in_use: HashSet<String> = HashSet::new();
+    let mut artifacts_in_use: HashSet<String> = HashSet::new();
+    let existing_main_crates: Vec<String> = existing_pipeline_ids
         .iter()
-        .map(|(_, _, source_checksum, _)| source_checksum.clone())
+        .map(|pipeline_id| {
+            format!(
+                "feldera_pipe_pipeline_{}_main",
```

Comment:
This should be utility function main_crate_name().
Or maybe you want a struct for a pipeline which has lots of such functions, including main, global, etc.

Reply:
Agreed, as this is now used in many place, it seems worthwhile to make a reusable function/struct